src/snowflake/connector/auth_webbrowser.py

```diff
@@ -95,7 +96,9 @@ def authenticate(self, authenticator, service_name, account, user, password):
         socket_connection = self._socket(socket.AF_INET, socket.SOCK_STREAM)
         try:
             try:
-                socket_connection.bind(("localhost", 0))
+                socket_addr = os.getenv("SF_AUTH_SOCKET_ADDR", "localhost")
+                socket_port = int(os.getenv("SF_AUTH_SOCKET_PORT", 0))
+                socket_connection.bind((socket_addr, socket_port))
```

Comment:
Since these variables wouldn't be used more than once let's in-line them
```suggestion
                socket_connection.bind((
                    os.getenv("SF_AUTH_SOCKET_ADDR", "localhost"),
                    int(os.getenv("SF_AUTH_SOCKET_PORT", 0))
                ))
```